PatternSyntaxException in UniqueProperties when duplicated property key has dash inside brackets

I have read check documentation: https://checkstyle.org/checks/misc/uniqueproperties.html#UniqueProperties
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
[root@9c767d97c868 checkstyle]# cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="UniqueProperties"/>
</module>
```

```bash
[root@9c767d97c868 checkstyle]# cat example.properties
some.key[index-with-dash]=value
some.key[index-with-dash]=value
```

```bash
[root@9c767d97c868 checkstyle]# java -Duser.language=en -Duser.country=US -jar checkstyle-13.7.0-all.jar -c config.xml example.properties
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing example.properties
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:313)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:227)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:429)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:347)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:205)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:129)
Caused by: java.util.regex.PatternSyntaxException: Illegal character range near index 16
^some.key[index-with-dash][\s:=].*$
                ^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2204)
        at java.base/java.util.regex.Pattern.range(Pattern.java:2999)
        at java.base/java.util.regex.Pattern.clazz(Pattern.java:2887)
        at java.base/java.util.regex.Pattern.sequence(Pattern.java:2315)
        at java.base/java.util.regex.Pattern.expr(Pattern.java:2245)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1945)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1576)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1101)
        at com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.getKeyPattern(UniquePropertiesCheck.java:148)
        at com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.getLineNumber(UniquePropertiesCheck.java:119)
        at com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.processFiltered(UniquePropertiesCheck.java:101)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:101)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:341)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:300)
        ... 5 more
Checkstyle ends with 1 errors.
```

Seems to be caused by `getKeyPattern` not escaping `keyName` properly so some regex values can sneak in while legal in properties file.

https://github.com/checkstyle/checkstyle/blob/be240059e1505302dc6dd46802af919411327d89/src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java#L145-L149

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java
@@ -143,8 +143,11 @@ public class UniquePropertiesCheck extends AbstractFileSetCheck {
      * @return regular expression pattern given key name
      */
     private static Pattern getKeyPattern(String keyName) {
-        final String keyPatternString = "^" + SPACE_PATTERN.matcher(keyName)
-                .replaceAll(Matcher.quoteReplacement("\\\\ ")) + "[\\s:=].*$";
+        final String keyNameWithEscapedSpaces = SPACE_PATTERN.matcher(keyName)
+                .replaceAll(Matcher.quoteReplacement("\\ "));
+        final String keyPatternString = "^"
+                + Pattern.quote(keyNameWithEscapedSpaces)
+                + "[\\s:=].*$";
         return Pattern.compile(keyPatternString);
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java
@@ -170,6 +170,15 @@ public class UniquePropertiesCheckTest extends AbstractModuleTestSupport {
             .isEqualTo(expected2);
     }
 
+    @Test
+    public void testRegexMetaCharacters() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(UniquePropertiesCheck.class);
+        final String[] expected = {
+            "1: " + getCheckMessage(MSG_KEY, "some.key[index-with-dash]", 2),
+        };
+        verify(checkConfig, getPath("InputUniquePropertiesRegex.properties"), expected);
+    }
+
     /**
      * Method generates NoSuchFileException details. It tries to open a file that does not exist.
      *

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/uniqueproperties/InputUniquePropertiesRegex.properties
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/uniqueproperties/InputUniquePropertiesRegex.properties
@@ -1,0 +1,2 @@
+some.key[index-with-dash]=value
+some.key[index-with-dash]=value

